DirectProgramming/DPC++/StructuredGrids/guided_HSOpticalflow_SYCLMigration/README.md

```diff
@@ -0,0 +1,167 @@
+> - For private installations: `. ~/intel/oneapi/setvars.sh`
+> - For non-POSIX shells, like csh, use the following command: `$ bash -c 'source <install-dir>/setvars.sh ; exec csh'`
+>
+>For more information on environment variables, see [Use the setvars Script with Linux* or macOS*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html), or [Windows](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html).
+
+### On Linux*
+Perform the following steps:
+1. Change to the `HSOpticalFlow` directory.
+2. Build the program.
+   ```
+   $ mkdir build
+   $ cd build
+   $ cmake ..
+   $ make
+   ```
+
+   By default, these commands build the `02_sycl_dpct_migrated`, `03_sycl_migrated` and `04_sycl_migrated_optimized` versions of the program.
+
+If an error occurs, you can get more details by running `make` with the `VERBOSE=1` argument:
+```
+make VERBOSE=1
+```
+
+#### Troubleshooting
+If you receive an error message, troubleshoot the problem using the Diagnostics Utility for Intel&reg; oneAPI Toolkits. The diagnostic utility provides configuration and system checks to help find missing dependencies, permissions errors and other issues. See [Diagnostics Utility for Intel&reg; oneAPI Toolkits User Guide](https://www.intel.com/content/www/us/en/develop/documentation/diagnostic-utility-user-guide/top.html).
+
+## Run the `HSOpticalFlow` Sample
+In all cases, you can run the programs for CPU and GPU. The run commands indicate the device target.
+1. Run `02_sycl_dpct_migrated` for CPU and GPU.
+    ```
+    make run_sdm_cpu
+    make run_sdm_gpu
+    ```
+    
+2. Run `03_sycl_migrated` for CPU and GPU.
+    ```
+    make run_cpu
+    make run_gpu
+    ```
+    
+3. Run `04_sycl_migrated_optimized` for CPU and GPU.
+    ```
+    make run_smo_cpu
+    make run_smo_gpu
+    ```
+
+### Run the `HSOpticalFlow` Sample in Intel&reg; DevCloud
+
+When running a sample in the Intel&reg; DevCloud, you must specify the compute node (CPU, GPU, FPGA) and whether to run in batch or interactive mode. For more information, see the Intel&reg; oneAPI Base Toolkit [Get Started Guide](https://devcloud.intel.com/oneapi/get_started/).
+
+#### Build and Run Samples in Batch Mode (Optional)
+
+You can submit build and run jobs through a Portable Bash Script (PBS). A job is a script that submitted to PBS through the `qsub` utility. By default, the `qsub` utility does not inherit the current environment variables or your current working directory, so you might need to submit jobs to configure the environment variables. To indicate the correct working directory, you can use either absolute paths or pass the `-d \<dir\>` option to `qsub`.
+
+1. Open a terminal on a Linux* system.
+2. Log in to Intel&reg;7 DevCloud.
```

Comment:
remove "7"

Reply:
Deleted a.out files and removed "7" from Readme.